internal/configs/sharding/instance.go

```diff
@@ -37,18 +37,26 @@ type instance struct {
 	slotsLimit                      int // HIP-16: The absolute number of maximum effective slots per shard limit for each validator. 0 means no limit.
 	allowlist                       Allowlist
 	feeCollectors                   FeeCollectors
+	emissionFraction                numeric.Dec
+	recoveryAddress                 ethCommon.Address
 }
 
 type FeeCollectors map[ethCommon.Address]numeric.Dec
 
 // NewInstance creates and validates a new sharding configuration based
 // upon given parameters.
 func NewInstance(
-	numShards uint32, numNodesPerShard, numHarmonyOperatedNodesPerShard, slotsLimit int, harmonyVotePercent numeric.Dec,
+	numShards uint32,
+	numNodesPerShard,
+	numHarmonyOperatedNodesPerShard,
+	slotsLimit int,
+	harmonyVotePercent numeric.Dec,
 	hmyAccounts []genesis.DeployAccount,
 	fnAccounts []genesis.DeployAccount,
 	allowlist Allowlist,
 	feeCollectors FeeCollectors,
+	emissionFractionToRecovery numeric.Dec,
+	recoveryAddress ethCommon.Address,
```

Comment:
why is recovery address not a configuration as part of sharding config like here https://github.com/harmony-one/harmony/blob/1b9614ba240de1fb431c8f800bb378f2358ae7eb/internal/configs/sharding/mainnet.go#L52

Reply:
let's leave this here as I am thinking  we may in the future change to another address eventually.